examples/morpheusvm/controller/controller.go

```diff
@@ -28,28 +28,28 @@ var (
 )
 
 // New returns a VM with the indexer, websocket, and rpc apis enabled.
-func New(options ...vm.Option) *vm.VM {
-	opts := []vm.Option{
-		indexer.WithIndexer(consts.Name, indexer.Endpoint),
-		ws.WithWebsocketAPI(10_000_000),
-		vm.WithVMAPIs(jsonrpc.JSONRPCServerFactory{}),
-		vm.WithControllerAPIs(&jsonRPCServerFactory{}),
+func New(namespacedOptions ...vm.NamespacedOption) *vm.VM {
+	opts := []vm.NamespacedOption{
+		vm.NewNamespacedOption("indexer", indexer.WithIndexer(consts.Name, indexer.Endpoint)),
+		vm.NewNamespacedOption("ws", ws.WithWebsocketAPI()),
+		vm.NewNamespacedOption("vmAPI", vm.WithVMAPIs(jsonrpc.JSONRPCServerFactory{})),
+		vm.NewNamespacedOption("controllerAPI", vm.WithControllerAPIs(&jsonRPCServerFactory{})),
 	}
```

Comment:
Can we change from naming them here to each service providing its own name? Each can export both the option function and the namespace, so that it can be overridden if desired, but this will remove the responsibility from the controller to define the namespace

Reply:
done